### PR TITLE
Plat 542 public api

### PR DESCRIPTION
**Pull request by phanyzewski**
Adds public api for secrets management

### CODE REVIEW

File: pkg/vault/vault.go [hunk 1]
```diff
@@ -53,6 +53,98 @@ func GetSecrets(ctx context.Context, secretValues *map[string]map[string]string,
...
+
+	config, err := loadVaultEnvironment()
+	if err != nil {
+		return fmt.Errorf("load client environment: %w", err)
+	}
+
+	vc, err := NewVaultClient(ctx, config)
+	if err != nil {
+		return fmt.Errorf("error initializing vault client: %w", err)
+	}
```
Comment (timdsnap) [inline pkg/vault/vault.go:76]: This is not DRY with the two functions below.  Can we clean up this repetition?

Comment (phanyzewski) [reply pkg/vault/vault.go]: I am pretty close to removing logging from here entirely which would help clean it up even more.

File: pkg/vault/vault_test.go [hunk 2]
```diff
@@ -28,6 +34,65 @@ func TestGetSecretVersions(t *testing.T) {
...
+	err := vault.GetSecretVersions(context.Background(), res, []string{"my-key"})
 	is.NoErr(err)
 }
+
+func TestCreateSecret(t *testing.T) {
+	is := is.New(t)
+	loginServer := vaultLoginServer()
+	defer loginServer.Close()
+
+	os.Setenv("VAULT_ADDR", loginServer.URL)
```
Comment (timdsnap) [inline pkg/vault/vault_test.go:46]: There's repetition here too, but I usually dont mind it in tests.

**Review by timdsnap — APPROVED** [review]
LGTM, feel free to ignore the one test comment.

**Comment by timdsnap** [comment]
Is there a reason we didn't use the jira ticket in the commit messages?  just curious really, haven't seen that pattern before

**Comment by phanyzewski** [comment]
> Is there a reason we didn't use the jira ticket in the commit messages? just curious really, haven't seen that pattern before

Nope, I'm just getting used to the semantic commits myself and probably could have added them.  On the other hand since this is an oss project linking to jira tickets doesn't make a ton of sense.

**Pull request merged**